tests/ui/vec_init_then_push.rs

```diff
@@ -44,3 +44,45 @@ pub fn no_lint() -> Vec<i32> {
         }
     }
 }
+
+fn _from_iter(items: impl Iterator<Item = u32>) -> Vec<u32> {
+    let mut v = Vec::new();
+    v.push(0);
+    v.push(1);
+    v.extend(items);
+    v
+}
```

Comment:
Could you add a second test like this, that reaches the threshold and triggers the lint? From my understanding, it should be as simple as:

```rs
fn _from_iter_trigger(items: impl Iterator<Item = u32>) -> Vec<u32> {
    let mut v = Vec::new();
    v.push(0);
    v.push(1);
    v.push(2);
    v.push(3);
    v.extend(items);
    v
}
```